### PR TITLE
1815 Handle locked (or not locked) materialized views better functionality

### PR DESCRIPTION
**Pull request by Ferrisx4**
# Bug Fix

## Description

This PR adds or modifies several functionalities related to custom tables and materialized views:

### Visual
 1. Make "Populate" the default action on the materialized view list; edit is a dropdown action now.
 2. The locked status of custom tables and materialized views is now shown in their respective listings on `admin/tripal/storage/chado/custom_tables` and `admin/tripal/storage/chado/mviews` 
Screen Shot 2024-05-14 at 10 37 38 AM

 3. On the Custom Tables page, add a blurb about what these are and how to find their respective materialized views.
 4. On the Custom Tables and Materialized Views lists, the filter now has a description. 
Screen Shot 2024-05-14 at 10 35 28 AM

### Functional
 1. When adding a new materialized view, do not allow the form to submit if the materialized view already exists (uniqueness determined by combination of table name and schema). 
 2. Disallow editing to custom tables and their respective materialized view if the table is locked.
 3. A materialized view can now be locked at creation time in the GUI.

## Testing?

 - The visual changes introduced in this PR can be tested by navigating to the pages above in item 5.
 - To test functional changes, create a new table by using the GUI, then create a materialized view based on that table. 
   - If you set the mview to be locked when you created it, you should find that you can no longer edit it. 
   - If you try to re-add the same materialized view (with the same table name and schema), the page should give you a helpful warning message. 
   - If you try to edit a materialized view that is locked, the Submit button should be disabled and there should be a status message at the top of the form indicating that it cannot be edited.

### CODE REVIEW

**Comment by Ferrisx4** [comment]
Looks like I somehow forgot to follow the branch naming guide this time, oops! I'll resubmit.

File: tripal_chado/src/Form/ChadoMviewForm.php [hunk 1]
```diff
@@ -289,6 +313,16 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
     foreach ($errors as $error) {
       $form_state->setErrorByName('schema', $error);
     }
+
+    // When adding a new materialized view, make sure that it is not an existing materialized view. We check for uniqueness by combining table name and schema, as some sites may have more than one instance of Chado installed.
```
Comment (spficklin) [inline tripal_chado/src/Form/ChadoMviewForm.php]:
```suggestion
// When adding a new materialized view, make sure that it is not an existing 
// materialized view.  We check for uniqueness by combining table name and 
// schema, as some sites may have more than one instance of Chado installed,
```

Comment (spficklin) [reply tripal_chado/src/Form/ChadoMviewForm.php]: Comments shouldn't go past character 80.

**Review by spficklin — COMMENTED** [review]
I think @Ferrisx4 that your code must not all be pushed. I'm not seeing the changes shown in the screenshot. There is no "locked" column in the mviews page, and the action doesn't default to "populate" like it shows.

**Comment by Ferrisx4** [comment]
Oops, I had word wrap enabled in my editor, didn't realize the length of the comment. Thanks for the quick fix @spficklin 

For the display issues, the changes on the Custom Tables and Materialized Views views are not picked up automatically. If you did not reinstall Tripal, they will not appear.

**Review by laceysanderson — CHANGES_REQUESTED** [review]
> For the display issues, the changes on the Custom Tables and Materialized Views views are not picked up automatically. If you did not reinstall Tripal, they will not appear.

You will want to add an update hook then. It looks like the changes are all config related 🤔 and I don't know off my head how to trigger a rediscover of that so you'll need to do some searching.

**Comment by Ferrisx4** [comment]
Ok thanks @laceysanderson I will look into it!

**Comment by spficklin** [comment]
@Ferrisx4, I know how to trigger a reload of config files.  Message me on Slack and I'll show you how.

**Review by laceysanderson — COMMENTED** [review]
I did some testing on a docker built on this branch and came across the following:

I get the following error when I edit an unlocked materialized view:
Screenshot 2024-06-22 at 3 27 34 PM

The following worked as expected:
- locked column was filled as expected
- default actions change as expected
- editing a locked materialized view was not allowed
- editing a locked custom table was not allowed
- editing an unlocked custom table was successful without error.
- action buttons worked as expected.

**Review by laceysanderson — COMMENTED** [review]
I also build a new docker on 4.x, confirmed that the changes in this PR were not seen, then switched to this branch and cleared the cache. At this point the changed did appear as expected ✅  This shows that the error Stephen first found showing that the config had not been updated has been fixed by the additions to the rebuild command in this PR.

**Comment by Ferrisx4** [comment]
Hi @laceysanderson, I believe I've fixed the error you found above. The form['locked'] form element was only available when the form's action was set to "add" and not when it was "edit". When editing, the locked value is now picked up from the database and saved in a 'value' form element to make it available for submission.

**Review by laceysanderson — APPROVED** [review]
Perfect! Thanks, this is ready to merge now 🎉

**Pull request merged**